src/main/java/org/broadinstitute/hellbender/utils/SequenceDictionaryUtils.java

```diff
@@ -87,6 +89,11 @@ public static void validateDictionaries( final String name1,
             case IDENTICAL:
                 return;
             case COMMON_SUBSET:
+                if (isReadsToReferenceComparison) {
+                    throw new UserException.IncompatibleSequenceDictionaries("Reference is missing contigs found in reads", name1, dict1, name2, dict2);
+                }
+                return;
+            case TWO_SUPERSETS_ONE:
                 return;
             case NO_COMMON_CONTIGS:
```

Comment:
You need to make sure that if we get `DIFFERENT_INDICES` and we're in a reads to reference comparison, we always throw (since we want to require `IDENTICAL` or `SUPERSET` for reads + ref comparisons now.